### PR TITLE
CWMS-866 Feature/o auth2 auto refresh

### PR DESCRIPTION
**Pull request by rma-bryson**
Removing unused code-gen annotation

### CODE REVIEW

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java [hunk 1]
```diff
@@ -39,10 +39,15 @@ public synchronized Request authenticate(Route route, Response response) throws
...
+        return newRequestWithAccessTokenAsHeader(response, updatedToken);
+    }
+
+    //package scoped for testing
+    Request newRequestWithAccessTokenAsHeader(Response response, OAuth2Token oauth2Token) {
         return response.request()
             .newBuilder()
             .removeHeader(AUTHORIZATION_HEADER)
-            .addHeader(AUTHORIZATION_HEADER, "Bearer " + updatedToken.getAccessToken())
+            .addHeader(AUTHORIZATION_HEADER, oauth2Token.getTokenType() + " " + oauth2Token.getAccessToken())
```
Comment (rma-psmorris) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java:50]: Assuming we can always put the token into the header? This is valid, just asking

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/auth/OAuth2TokenProvider.java [hunk 2]
```diff
@@ -7,4 +7,10 @@ public interface OAuth2TokenProvider {
     OAuth2Token getToken() throws IOException;
 
     OAuth2Token refreshToken() throws IOException;
+
+    String getKeyForTokenIfParameter();
+
+    default boolean useAuthorizationHeader() {
+        return getKeyForTokenIfParameter() != null;
```
Comment (rma-psmorris) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/auth/OAuth2TokenProvider.java:14]: Is this condition correct? The way I read it is that I should use the Authorization Header method if I am setup to use the token as a parameter.

If I don't have the token as a parameter, then I should not use the authorization header method.

File: cwms-http-client/src/test/java/mil/army/usace/hec/cwms/http/client/TestOAuth2TokenInterceptor.java [hunk 3]
```diff
@@ -52,6 +55,11 @@ public OAuth2Token refreshToken() {
                 token.setRefreshToken("456xyz");
                 return token;
             }
+
+            @Override
+            public String getKeyForTokenIfParameter() {
+                return null;
```
Comment (rma-psmorris) [inline cwms-http-client/src/test/java/mil/army/usace/hec/cwms/http/client/TestOAuth2TokenInterceptor.java:61]: Where is the not null test?

**Review by rma-psmorris — COMMENTED** [review]
Questions

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java (hunk 1 above)
Comment (MikeNeilson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java:50]: If someone gives you a system where the Token *DOESN'T* go in a header... run.

Unless you're thinking of something else?

File: cwms-http-client/src/test/java/mil/army/usace/hec/cwms/http/client/TestOAuth2TokenInterceptor.java (hunk 3 above)
Comment (rma-bryson) [reply cwms-http-client/src/test/java/mil/army/usace/hec/cwms/http/client/TestOAuth2TokenInterceptor.java]: This was also removed.

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java (hunk 1 above)
Comment (rma-psmorris) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java:50]: RMA has worked with a USBR service that had it as a parameter, so I asked Bryson to make sure his design allows for that but to not implement it currently as there are no CWMS web services that require that.

Comment (rma-bryson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java:50]: I would say that if we need to provide an Authenticator that adds it as a parameter, then we can implement an Authenticator that does that, when needed.

Comment (MikeNeilson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java:50]: > RMA has worked with a USBR service that had it as a parameter, so I asked Bryson to make sure his design allows for that but to not implement it currently as there are no CWMS web services that require that.

... And *THAT* is how credentials leak. We won't be putting credentials in query parameters... ever. (for our services.)

**Review by rma-psmorris — APPROVED** [review]
no further comments

**Pull request merged**